src/DETHRACE/common/graphics.c

```diff
@@ -792,9 +792,48 @@ void CalculateWobblitude(tU32 pThe_time) {
     double cosine_over_angle;
     LOG_TRACE("(%d)", pThe_time);
 
+    if (gProgram_state.new_view != eView_undefined) {
+        return;
+    }
     gScreen_wobble_x = 0;
     gScreen_wobble_y = 0;
-    STUB_ONCE();
+    for (i = 0; i < COUNT_OF(gWobble_array); i++) {
+        if (gWobble_array[i].time_started != 0) {
+            time_going = pThe_time - gWobble_array[i].time_started;
+            if (time_going > 1000) {
+                gWobble_array[i].time_started = 0;
+            } else {
+                mod_angle = fmod(time_going / gWobble_array[i].period, TAU);
+                if (mod_angle > DR_3PI_OVER_2) {
+                    cosine_over_angle = gCosine_array[(unsigned int)((TAU - mod_angle) / DR_PI * 128.0)];
+                } else if (mod_angle > DR_PI) {
+                    cosine_over_angle = -gCosine_array[(unsigned int)((mod_angle - DR_PI) / DR_PI * 128.0)];
+                } else if (mod_angle > DR_PI_OVER_2) {
+                    cosine_over_angle = -gCosine_array[(unsigned int)((DR_PI - mod_angle) / DR_PI * 128.0)];
+                } else {
+                    cosine_over_angle = gCosine_array[(unsigned int)(mod_angle / DR_PI * 128.0)];
+                }
+                angle = cosine_over_angle / ((double)(pThe_time - gWobble_array[i].time_started) * 0.0035f + 1.0f);
+                gScreen_wobble_x = (gWobble_array[i].amplitude_x * angle + gScreen_wobble_x);
+                gScreen_wobble_y = (gWobble_array[i].amplitude_y * angle + gScreen_wobble_y);
+            }
+        }
+    }
+    if (gCurrent_graf_data->cock_margin_x >= gScreen_wobble_x) {
+        if (-gCurrent_graf_data->cock_margin_x > gScreen_wobble_x) {
+            gScreen_wobble_x = -gCurrent_graf_data->cock_margin_x;
+        }
+    } else {
+        gScreen_wobble_x = gCurrent_graf_data->cock_margin_x;
+    }
+    if (gCurrent_graf_data->cock_margin_y >= gScreen_wobble_y) {
+        if (-gCurrent_graf_data->cock_margin_y > gScreen_wobble_y) {
+            gScreen_wobble_y = -gCurrent_graf_data->cock_margin_y;
+        }
+    } else {
+        gScreen_wobble_y = gCurrent_graf_data->cock_margin_y;
+    }
```

Comment:
A `CLAMP` macro would be nice. But probably out of scope for the dethrace project.

```suggestion
    if (gScreen_wobble_x > gCurrent_graf_data->cock_margin_x) {
        gScreen_wobble_x = gCurrent_graf_data->cock_margin_x;
    } else if (gScreen_wobble_x < -gCurrent_graf_data->cock_margin_x) {
        gScreen_wobble_x = -gCurrent_graf_data->cock_margin_x;
    }
    if (gScreen_wobble_y > gCurrent_graf_data->cock_margin_y) {
        gScreen_wobble_y = gCurrent_graf_data->cock_margin_y;
    } else if (gScreen_wobble_y < -gCurrent_graf_data->cock_margin_y) {
        gScreen_wobble_y = -gCurrent_graf_data->cock_margin_y;
    }
```